test/content-negotiation-test.js

```diff
@@ -0,0 +1,61 @@
+var test = require('tape');
+var Hapi = require('hapi');
+var server = new Hapi.Server();
+var routes = require('../routes.js');
+
+server.connection({
+  port: process.env.TESTPORT || '8080'
+});
+
+server.route(routes);
+
+server.register(require('hapi-negotiator'), function (err) {
+  if (err) {
+    console.error('Failed to load plugin:', err);
+  }
+});
```

Comment:
We now need to refactor our `server.js` file to return a function that takes config and a callback as its argument. We can then reuse it here to test the actual server code.

e.g.

**server.js**
```js
module.exports = (config, cb) => {
  const server = new Hapi.Server()

  server.connection({ port: config.port })
  server.route(routes)

  server.register([require('inert'), require('hapi-negotiator')], (err) => {
    if (err) return cb(err)
    cb(null, server)
  })
}
```

I suggest we create a `testWithServer` helper module that will create a new server for us to test against, and we might use it like this:

```js
testWithServer('Should do something', (t, server) => {
  // ...
})
```

For running the server we need a `bin/server.js` which will setup and start a new server for us:

```js
const config = require('../config')
const createServer = require('../server')

createServer(config, (err, server) => {
  if (err) throw err

  server.start(() => {
    if (err) throw err
    console.log(server.info.uri)
  })
})
```
The `npm start` script will need to be changed to call this file instead.